src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -0,0 +1,265 @@
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final String isoCountryCode = edge.tag(ISOCountryTag.KEY).toUpperCase();
+
+        // Get all edges in the roundabout
+        final List<Edge> roundaboutEdges = getAllRoundaboutEdges(edge);
+
+        // Get the direction of the roundabout
+        final RoundaboutDirection direction = findRoundaboutDirection(roundaboutEdges);
+
+        // Determine if the roundabout is in a left or right driving country
+        final boolean isLeftDriving = leftDrivingCountries.contains(isoCountryCode);
+
+        // If the roundabout is found to be going in multiple directions
+        if (direction.equals(RoundaboutDirection.MULTIDIRECTIONAL))
+        {
+            return Optional.of(this.createFlag(new HashSet<>(roundaboutEdges),
+                    this.getLocalizedInstruction(1, edge.getOsmIdentifier())));
+        }
+        // If the roundabout traffic is clockwise in a right-driving country, or
+        // If the roundabout traffic is counterclockwise in a left-driving country
+        if (direction.equals(RoundaboutDirection.CLOCKWISE) && !isLeftDriving
+                || direction.equals(RoundaboutDirection.COUNTERCLOCKWISE) && isLeftDriving)
+        {
+            return Optional.of(this.createFlag(new HashSet<>(roundaboutEdges),
+                    this.getLocalizedInstruction(0, edge.getOsmIdentifier())));
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * This method gets all edges in a roundabout given one edge in that roundabout, in ascending
+     * Edge identifier order.
+     *
+     * @param edge
+     *            An Edge object known to be a roundabout edge
+     * @return A list of edges in the roundabout
+     */
+    private List<Edge> getAllRoundaboutEdges(final Edge edge)
+    {
+        final List<Edge> roundaboutEdges = new ArrayList<>();
+
+        // Initialize a queue to add yet to be processed connected edges to
+        final Queue<Edge> queue = new LinkedList<>();
+
+        // Mark the current Edge as visited and enqueue it
+        this.markAsFlagged(edge.getIdentifier());
+        queue.add(edge);
+
+        // As long as the queue is not empty
+        while (!queue.isEmpty())
+        {
+            // Dequeue a connected edge and add it to the roundaboutEdges
+            final Edge currentEdge = queue.poll();
+
+            roundaboutEdges.add(currentEdge);
+
+            // Get the edges connected to the edge as an iterator
+            final Set<Edge> connectedEdges = currentEdge.connectedEdges();
+
+            for (final Edge connectedEdge : connectedEdges)
+            {
+                final Long edgeId = connectedEdge.getIdentifier();
```

Comment:
This is only needed if condition for the `if` block below evaluates to true. I'd suggest removing it and just calling `this.markAsFlagged(connectedEdge.getIdentifier())`.